src/main/java/com/khutircraftubackend/product/ProductController.java

```diff
@@ -13,69 +15,80 @@
-                                                      @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse patchProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
 
         ProductEntity patchedProduct = productService.patchProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(patchedProduct);
+        return productMapper.toProductResponse(patchedProduct);
     }
 
     @PutMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<ProductEntity> updateProduct(@PathVariable Long productId,
-                                                       @ModelAttribute ProductUpdateRequest request,
-                                                       @RequestParam("thumbnailImage") MultipartFile thumbnailImage,
-                                                       @RequestParam("images") MultipartFile image) throws IOException {
+    @ResponseStatus(HttpStatus.OK)
+    public ProductResponse updateProduct(
+            @Valid
+            @PathVariable Long productId,
+            @ModelAttribute ProductUpdateRequest request,
+            @RequestPart(value = "thumbnailImage", required = false) MultipartFile thumbnailImage,
+            @RequestPart(value = "image", required = false) MultipartFile image) throws IOException {
+
         ProductEntity updatedProduct = productService.updateProduct(productId, request, thumbnailImage, image);
 
-        return ResponseEntity.ok(updatedProduct);
+        return productMapper.toProductResponse(updatedProduct);
     }
 
     @DeleteMapping("/{productId}")
     @PreAuthorize("hasRole('SELLER') and @productService.canModifyProduct(#productId)")
-    public ResponseEntity<Void> deleteProduct(@PathVariable Long productId) throws IOException {
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteProduct(@PathVariable Long productId) throws IOException {
+
         productService.deleteProduct(productId);
-        return ResponseEntity.noContent().build();
+
     }
 
     @DeleteMapping("/delete-all")
     @PreAuthorize("hasRole('SELLER')")
-    public ResponseEntity<Void> deleteAllProductsForCurrentSeller() throws IOException {
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteAllProductsForCurrentSeller() throws IOException {
+
         productService.deleteAllProductsForCurrentSeller();
-        return ResponseEntity.noContent().build();
+
     }
 
     @GetMapping
     public List<ProductEntity> getAllProducts(
```

Comment:
oops, 
probably fogot annotate with `@ResponseStatus(HttpStatus.OK)`

Reply:
I didn't forget. Didn't know.

Reply:
Why return the status?

Reply:
although, it's gonna work without it, 
this is the same thing/identical to sending status with ResponseEntity:
- `return ResponseEntity.ok()....`
- `return ResponseEntity.status(HttpStatus.OK)...`

just the same thing, but coded differently, with an alternative: `@ResponseStatus`

please check these links out:
- https://mnachit.medium.com/ed7d478023e8
- https://www.baeldung.com/spring-response-entity

But Spring documentation suggests using [@ResponseBody](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-methods/responsebody.html) for method that return value and `@ResponseStatus` for methods that a void: [Return Values](https://docs.spring.io/spring-framework/reference/web/webflux/controller/ann-methods/return-types.html).

So following best Spring practices, it would be actually better to apply `@ResponseBody`.

Reply:
oh, btw, to make decision process easier, please test responses from this controller in Postman, see what responses do you get ;)

Reply:
👍  
you see this? 
```json
    "user": {
        "id": 2,
        "email": "carinechekanova@gmail.com",
        "password": "$2a$10$4xfs1lBQkC7YSvjpXm7s/esN6jNNZljlWhooZ14OPffNQvHjDorBa",
        "role": "SELLER",
        "enabled": true,
        "creationDate": "2024-10-14T11:29:01.475862",
        "confirmationToken": null
    },
```
that "user" data strictly should not be present in any of responses. 
if we provide in any of our responses - this is a security breach.
needs to be fixed.

Reply:
This is not my jira ticket. So the question is not for me, I just emphasized the problem.

Reply:
1. you are developing Product Service and your code is returning this:
```java
public record ProductResponse(
        Long id,
        String name,
        String thumbnailImageUrl,
        String imageUrl,
        boolean available,
        String description,
        SellerEntity seller,           // <------ this is Entity, but supposed to be DTO or something safe
        CategoryEntity category        // <------ this is Entity, but supposed to be DTO or something safe
) {
}
```
2. Once code is merged into main branch is not someone else's code or jira ticket. this OUR code or jira ticket.

Reply:
1. Provide requirements about fields that should not be returned.
a). Currently, the password field is not displayed, other fields are in question.
2. When using @RestController, there is no need to additionally use @ResponseBody.

Reply:
> Provide requirements about fields that should not be returned.

none of the data from UserEntity should be present in any of responses. UserEntity is only used internally. This is a secure representation of the client/user/customer. 

> When using `@RestController`, there is no need to additionally use `@ResponseBody`.

ok, agree with that, coz as they say here [@ResponseBody](https://docs.spring.io/spring-framework/reference/web/webmvc/mvc-controller/ann-methods/responsebody.html): `@RestController = @Controller + @ResponseBody.`

so, back to your question
> Why return the status?

as i said, it's gonna work without it, but if you don't really want it, i don't care.
but, this is a common pattern that you can find in repositories such as `Microsoft` and `micronaut-projects`
here: https://github.com/search?q=%40ResponseStatus%28HttpStatus.OK%29&type=code

Reply:
Got it, thanks.